homeassistant/components/photo_frame/image.py

```diff
@@ -0,0 +1,124 @@
+"""Support for Photo Frame image."""
+
+from __future__ import annotations
+
+from dataclasses import dataclass
+import logging
+from pathlib import Path
+import random
+
+from homeassistant.components.image import ImageEntity, ImageEntityDescription
+from homeassistant.components.media_player import MediaClass
+from homeassistant.components.media_source import (
+    async_browse_media,
+    async_resolve_media,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import EVENT_HOMEASSISTANT_STARTED
+from homeassistant.core import CoreState, HomeAssistant
+from homeassistant.helpers.entity_platform import AddConfigEntryEntitiesCallback
+from homeassistant.util import dt as dt_util
+
+_LOGGER = logging.getLogger(__name__)
+
+
+@dataclass(frozen=True, kw_only=True)
+class PhotoFrameImageEntityDescription(ImageEntityDescription):
+    """Photo Frame image entity description."""
+
+    media_content_id: str
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: ConfigEntry,
+    async_add_entities: AddConfigEntryEntitiesCallback,
+) -> None:
+    """Set up the Photo Frame image entities."""
+    if media := entry.data.get("media"):
+        description = PhotoFrameImageEntityDescription(
+            key="image",
+            name=entry.title,
+            media_content_id=media.get("media_content_id"),
+        )
+        async_add_entities(
+            [
+                PhotoFrameImageEntity(
+                    description,
+                    unique_id=entry.entry_id,
+                    hass=hass,
+                )
+            ]
+        )
+
+
+class PhotoFrameImageEntity(ImageEntity):
+    """Implement the image entity for Photo Frame."""
+
+    entity_description: PhotoFrameImageEntityDescription
+    _attr_content_type = "image/png"
+    path: Path | None
+
+    def __init__(
+        self,
+        description: PhotoFrameImageEntityDescription,
+        unique_id: str,
+        hass: HomeAssistant,
+    ) -> None:
+        """Initialize the entity."""
+        ImageEntity.__init__(self, hass)
+        self.entity_description = description
+        self.path = None
+        self.hass = hass
+        self._attr_unique_id = unique_id
+        self._attr_name = (
+            description.name if isinstance(description.name, str) else None
+        )
+
+    async def get_next_image(self) -> None:
+        """Update the image entity with the next image from the source media."""
+        media = await async_browse_media(
+            self.hass, self.entity_description.media_content_id
+        )
```

Comment:
`async_browse_media` can throw here again. Maybe just log it and leave the current image. Or set it to None. Idk